libs/utils/android/benchmark.py

```diff
@@ -0,0 +1,289 @@
+        return parser.parse_args()
+
+
+    def _getBmConf(self):
+        if self.bm_conf is None:
+            msg = 'Benchmark subclasses must override the `bm_conf` attribute'
+            raise NotImplementedError(msg)
+
+        # Override default configuration with command line parameters
+        if self.args.android_device:
+            self.bm_conf['device'] = self.args.android_device
+        if self.args.android_home:
+            self.bm_conf['ANDROID_HOME'] = self.args.android_home
+        if self.args.results_dir:
+            self.bm_conf['results_dir'] = self.args.results_dir
+        if self.args.collect:
+            self.bm_collect = self.args.collect
+
+        # Override energy meter configuration
+        if self.args.iio_channel_map:
+            em = {
+                'instrument'  : 'acme',
+                'channel_map' : {},
+            }
+            for ch in self.args.iio_channel_map.split(','):
+                ch_name, ch_id = ch.split(':')
+                em['channel_map'][ch_name] = ch_id
+            self.bm_conf['emeter'] = em
+            self._log.info('Using ACME energy meter channels: %s', em)
+
+        # Override EM if energy collection not required
+        if 'energy' not in self.bm_collect:
+            try:
+                self.bm_conf.pop('emeter')
+            except:
+                pass
+
+        return self.bm_conf
+
+    def _getWorkload(self):
+        if self.bm_name is None:
+            msg = 'Benchmark subclasses must override the `bm_name` attribute'
+            raise NotImplementedError(msg)
+        # Get a referench to the worload to run
+        wl = Workload.getInstance(self.te, self.bm_name)
+        if wl is None:
+            raise ValueError('Specified benchmark [{}] is not supported'\
+                             .format(self.bm_name))
+        return wl
+
+    def _getBmParams(self):
+        if self.bm_params is None:
+            msg = 'Benchmark subclasses must override the `bm_params` attribute'
+            raise NotImplementedError(msg)
+        return self.bm_params
+
+    def _getBmCollect(self):
+        if self.bm_collect is None:
+            msg = 'Benchmark subclasses must override the `bm_collect` attribute'
+            self._log.warning(msg)
+            return ''
+        return self.bm_collect
+
+    def __init__(self):
+        """
+        Set up logging and trigger running experiments
+        """
+        LisaLogging.setup()
+        self._log = logging.getLogger('Benchmark')
+
+        self._log.info('=== CommandLine parsing...')
+        self.args = self._parseCommandLine()
+
+        self._log.info('=== TestEnv setup...')
+        self.bm_conf = self._getBmConf()
+        self.te = TestEnv(self.bm_conf)
+        self.target = self.te.target
+
+        self._log.info('=== Initialization...')
+        self.wl = self._getWorkload()
+        self.out_dir=self.te.res_dir
+        try:
+            self.benchmarkInit()
```

Comment:
The benchmarkInit function returns True/False and does not raise any exceptions. That results in an error of setting the governor to be ignored here and the test to continue.

I'm using 
```
if not self.benchmarkInit():
            self._log.warning('Benchmark initialization failed: execution aborted')
            return
```
to make it work properly, that is not to continue the test if the benchmarkInit functions fails (returns False) but also not to raise an exception that would stop with failure the entire test (that might contain scenarios for other governors that would work).